packages/mobile/src/components/inbox-unavailable-drawer/InboxUnavailableDrawer.tsx

```diff
@@ -0,0 +1,256 @@
+  callToAction: ChatPermissionAction,
+  user: User,
+  styles: ReturnType<typeof useStyles>
+) => {
+  switch (callToAction) {
+    case ChatPermissionAction.NONE:
+      return (
+        <>
+          {messages.noAction}
+          <UserBadges
+            user={user}
+            nameStyle={styles.callToActionText}
+            as={Text}
+          />
+        </>
+      )
+    case ChatPermissionAction.TIP:
+      return (
+        <>
+          {messages.tipGated(
+            <UserBadges
+              user={user}
+              nameStyle={styles.callToActionText}
+              as={Text}
+            />
+          )}
+        </>
+      )
+    case ChatPermissionAction.UNBLOCK:
+      return <>{messages.blockee}</>
+    default:
+      return null
+  }
+}
+
+export const InboxUnavailableDrawer = () => {
+  const dispatch = useDispatch()
+  const navigation = useNavigation()
+  const styles = useStyles()
+  const neutralLight2 = useColor('neutralLight2')
+
+  const { userId } = useSelector((state) => getData<'InboxUnavailable'>(state))
+  const user = useSelector((state) => getUser(state, { id: userId }))
+  const { callToAction } = useSelector((state) => getCanChat(state, userId))
+
+  const closeDrawer = useCallback(() => {
+    dispatch(
+      setVisibility({
+        drawer: 'InboxUnavailable',
+        visible: false
+      })
+    )
+  }, [dispatch])
+
+  const handleUnblockPress = useCallback(() => {
+    dispatch(unblockUser({ userId }))
+    closeDrawer()
+  }, [dispatch, userId, closeDrawer])
+
+  const handleLearnMorePress = useCallback(() => {
+    // TODO: Link to blog
+    closeDrawer()
+  }, [closeDrawer])
+
+  const handleTipPress = useCallback(() => {
+    dispatch(beginTip({ user, source: 'profile' }))
+    navigation.navigate('TipArtist')
+    closeDrawer()
+  }, [closeDrawer, dispatch, navigation, user])
+
+  const actionToButtonsMap = useMemo(() => {
+    return {
+      [ChatPermissionAction.NONE]: [
+        {
+          buttonTitle: messages.learnMore,
+          buttonPress: handleLearnMorePress,
+          buttonVariant: 'common'
+        }
+      ],
+      [ChatPermissionAction.TIP]: [
+        {
+          buttonTitle: messages.sendAudio,
+          buttonPress: handleTipPress,
+          buttonVariant: 'primary',
+          buttonIcon: IconTip,
+          buttonTextStyle: styles.buttonTextWhite
+        }
+      ],
+      [ChatPermissionAction.UNBLOCK]: [
+        {
+          buttonTitle: messages.unblockUser,
+          buttonPress: handleUnblockPress,
+          buttonVariant: 'primary',
+          buttonTextStyle: styles.buttonTextWhite
+        },
+        {
+          buttonTitle: messages.cancel,
+          buttonPress: closeDrawer,
+          buttonVariant: 'common'
+        }
+      ]
+    }
+  }, [
+    handleLearnMorePress,
+    handleTipPress,
+    styles.buttonTextWhite,
+    handleUnblockPress,
+    closeDrawer
+  ])
+
+  const content = mapActionToContent(callToAction, user!, styles)
```

Comment:
I think the `user!` is safe because we only ever navigate to this drawer from that user's profile screen. Open to ideas tho.

Reply:
I'd say have the fn accept undefined and render null instead of user badges if undefined